aggregators/aggregator_test.go

```diff
@@ -1145,7 +1145,10 @@ func BenchmarkAggregateBatchSerial(b *testing.B) {
 func BenchmarkAggregateBatchParallel(b *testing.B) {
 	b.ReportAllocs()
 	agg := newTestAggregator(b)
-	defer agg.Close(context.Background())
+	go func() {
+		agg.Run(context.Background())
+	}()
+	// defer agg.Close(context.Background())
```

Comment:
Just a heads up, the main benchmark should also be run with this change or both should be run without this change. I found that using `agg.Close` was pushing all the complexity of flushing the multiple batches we create in the benchmark result. My thinking for doing this change was that the `Close` will not be in hot path.